models/cost_per_serverless_task_run.sql

```diff
@@ -0,0 +1,151 @@
+{{ config(
+    materialized='incremental',
+    unique_key=['query_id', 'start_time'],
+) }}
+
+with
+stop_threshold as (
+    select max(end_time) as latest_ts
+    from {{ ref('stg_serverless_task_history') }}
+),
+
+// task_history has all runs, but we only want serverless runs
+serverless_runs as (
+    select
+        task_history.* exclude (query_id), // just to remove ambiguous column names
+        query_history.*,
+        task_history.database_name as task_database_name,
+        task_history.schema_name as task_schema_name,
+        name as task_name,
+        concat(task_database_name, '.', task_schema_name, '.', task_name) as task_fqn_name,
+        start_time as execution_start_time,
+    from {{ ref('stg_task_history') }} as task_history
+
+    inner join {{ ref('stg_query_history') }} as query_history
+        on task_history.query_id = query_history.query_id and query_history.start_time = task_history.query_start_time
```

Comment:
confirming that joining on `start_time=query_start_time` is safe?

Reply:
Yes! I think there was a misunderstanding between you and Teghan. He meant something else with start_time not matching (parent vs. children), and not what I think you understood (Snowflake claiming that two columns in different tables should match and they don't).